extensions/resource-deployment/src/services/tools/toolBase.ts

```diff
@@ -19,14 +20,45 @@ export abstract class ToolBase implements ITool {
 	abstract type: ToolType;
 	abstract homePage: string;
 	abstract autoInstallSupported: boolean;
-	public abstract install(): Promise<void>;
+	abstract installationCommands: Command[];
 	protected abstract getVersionFromOutput(output: string): SemVer | undefined;
-	protected abstract readonly versionCommand: string;
+	protected abstract readonly versionCommand: Command;
+
+	protected get installationPath(): Promise<string | null> {
+		return new Promise<string | null>((resolve, _reject) => {
+			resolve(null);
+		});
+	}
+
+	protected get installationSearchPaths(): (string | null)[] {
+		return [this.storagePath];
+	}
+
+	protected get downloadPath(): string {
+		return this.storagePath;
+	}
+
+	public get storagePath(): string {
+		return this._platformService.storagePath();
+	}
+
+	public get osType(): OsType {
+		return this._osType;
+	}
 
 	public get version(): SemVer | undefined {
 		return this._version;
 	}
 
+	public get fullVersion(): string | undefined {
+		if (this._version !== undefined) {
+			return this._version && this._version.version;
```

Comment:
isn't this._version definitely not undefined here? Shouldn't have a reason to have &&

Reply:
good catch. Fixed